packages/commonwealth/client/scripts/state/api/threads/toggleArchive.ts

```diff
@@ -14,15 +15,19 @@ const toggleThreadArchive = async ({
   threadId,
   isArchived,
 }: ToggleThreadArchiveProps) => {
-  return await axios.post(
-    `${app.serverUrl()}/threads/${threadId}/${
-      !isArchived ? 'archive' : 'unarchive'
-    }`,
+  const method = isArchived ? 'delete' : 'put';
+  return await axios[method](
+    `${app.serverUrl()}/threads/${threadId}/archive`,
     {
       data: {
         jwt: app.user.jwt,
         chain_id: chainId,
       } as any,
+    },
+    {
+      headers: {
```

Comment:
We don't use JWT headers -- providing it as part of the data alone is sufficient.

Reply:
> We don't use JWT headers -- providing it as part of the data alone is sufficient.

During the resolution of the merge topics, I noticed that some API logic had been updated. I attempted to just pass the JWT as part of the data (which is what I originally did), but that resulted in repeated "Unauthorized" errors. Placing the JWT in the header fixed the issue. Perhaps there is some aspect of the backend logic that will require an update? @rbennettcw

Reply:
You can pass the token as the `jwt` property in the request body or as the bearer token in the headers. Both should work.

Reply:
BTW, with the latest code, you can/should use the `PATCH /threads/:id` route. It supports archive/unarchive.

The request body would be `{ archive: boolean }`, you can set to `true` or `false` and it's returned as a timestamp or `null`.

For the patch route, you may get the `Unauthorized` error if you don't have permissions to perform the operation.

Reply:
Updated code to use the new thread editing logic. Thanks!